lazy_theta_star_b/src/lazy_theta_star_b.cpp

```diff
@@ -0,0 +1,529 @@
+                                if (data[m_id].closed == 1) {
+                                    data[m_id].closed = -1;
+                                    push_to_pq(m_id);
+                                }
+                            }
+                            continue;
+                        }
+                    }
+                }
+
+                last_id = curr_id;
+                do {
+                    curr_id = pq[1].pos_id;
+                    cx = data[curr_id].x;
+                    cy = data[curr_id].y;
+                } while (!isSafe(cx, cy));
+                binary_heap_del_min();
+                data[curr_id].closed = -1;
+
+            }
+            
+            if(pq.size() == 1){
+                RCLCPP_INFO(node_->get_logger(), "NO PATH POSSIBLE");
+            }
+
+        } else {
+            RCLCPP_INFO(node_->get_logger(), "STRAIGHT LINE PATH");
+            data.push_back({(dst.x), (dst.y), dist(cx, cy, src.x, src.y), 0, 0});
+            curr_id = 1;
+        }
+
+        RCLCPP_INFO(node_->get_logger(), "REACHED DEST  %i,   %i", dst.x, dst.y);
+
+        backtrace(&path, curr_id);
+
+        cout << data.size() << '\n';
+
+        cout << "Publishing the Path" << '\n';
+
+        //reset all the values
+        global_path = linearInterpolation(path, interpolation_dist);
+
+        geometry_msgs::msg::PoseStamped p_end;
+        p_end.pose.position.x = goal.pose.position.x;
+        p_end.pose.position.y = goal.pose.position.y;
+        p_end.header.stamp = node_->now();
+        p_end.header.frame_id = global_frame_;
+        global_path.poses.push_back(p_end);
+
+        return global_path;
+    }
+
+    void LazyThetaStarB::initializePosn() {
+        if (pq.size() != abs(sizeX * sizeY)) {
+            for (int i = 0; i < sizeY * sizeX; i++) {
+                posn.push_back(0);
+            }
+        } else {
+            for (int i = 0; i < sizeY * sizeX; i++) {
+                posn[i] = 0;
+            }
+        }
+    }
+
+    void LazyThetaStarB::addIndex(int cx, int cy, id index) {
+        posn[sizeX * cy + cx] = index;
+    }
+
+    id LazyThetaStarB::getIndex(int cx, int cy) {
+        return posn[sizeX * cy + cx];
+    }
+
+    void LazyThetaStarB::backtrace(vector<id> *waypt, id curr_id) {
+        vector<id> waypt_rev;
+        do {
+            waypt_rev.push_back(curr_id);
+            curr_id = data[curr_id].parentId;
+        } while (curr_id != 0);
+        waypt_rev.push_back(0);
+
+        for (int i = waypt_rev.size() - 1; i >= 0; i--) {
+            waypt->push_back(waypt_rev[i]);
+        }
+
+    }
+
+    void LazyThetaStarB::binary_heap_del_min() {
+        int hole = 1;
+        int succ = 2, size = pq.size() - 1, sz = size;
+        //the main bottom up part where it compares and assigns the smallest value to the hole
+        //remember that for this part the heap starts at index 1
+        while (succ < sz) {
```

Comment:
I think you should use a `std::priority_queue` so that you're not manually doing the heap management. I did some testing with this in my hybrid A* planner and found it to be faster. This could be a place of significant overhead.

Reply:
It is just a tad bit faster for large number of nodes, and also for some reason, priority queue just didn't gave the right values (ie the smallest f_cost) at the top.

Reply:
You can set the comparison https://github.com/SteveMacenski/navigation2/blob/nav2_smac_planner/smac_planner/include/smac_planner/a_star.hpp#L48 for priority queue

Reply:
(so if its opposite, then you can change the `<`)

Reply:
Will try it once again, and I know that it gives the largest value at the top.

Reply:
You could also mess with the cost functions to be small when better rather than merge so that smallest first is  better (that's the typical way of doing it)

Reply:
A distance heuristic would make it smaller = better = top if pile.

Reply:
I didn’t understand your point

Reply:
Do you get now? Sorry, unclear to me from that comment if this needs more discussion

Reply:
I don’t how is that being shown to you, but I have deleted those files already

Reply:
I had deleted this file already.